File: app/src/main/java/com/haishinkit/util/EventUtils.java
package com.haishinkit.util;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

import com.haishinkit.events.Event;

public final class EventUtils {
    private EventUtils() {
    }

    public static Map<String, Object> toMap(final Event event) {
        Object data = event.getData();
        if (data == null || !(data instanceof Map)) {
            return new HashMap<String, Object>();
        }
        return (Map<String, Object>) data;
    }
}

File: haishinkit/src/main/java/com/haishinkit/util/EventUtils.kt
package com.haishinkit.util

import com.haishinkit.events.Event
import java.util.HashMap

object EventUtils {
    fun toMap(event: Event): Map<String, Any> {
        val data = event.data
        if (data == null || data !is Map<*, *>) {
            return HashMap()
        }
        return data as Map<String, Any>
    }
}
